navigator_engine/tests/unit_tests/test_conditionals.py

```diff
@@ -83,3 +86,24 @@ def test_check_resource_key(mock_engine, resource_type, key, value, expected):
 def test_check_dataset_valid(resources, expected, mock_engine):
     with pytest.raises(TypeError, match='123'):
         conditionals.check_not_skipped(123, mock_engine)
+
+
+@pytest.mark.parametrize("checklist, dataframe, expected, raises_error", [
+    (['Adult male ART has 2020 data', 'Ped VS as 2020 data', 'Adult ART coverage never exceeds 100%'],
+     pd.read_csv('../test_data/test_spectrum_check.csv'), True, does_not_raise()),
```

Comment:
This is interesting - i did not know about `does_not_raise()`.  I expect this is still work in progress and you are trying to parameterise the same test to check that the results when run correctly are as expected (you aren't checking this yet) and the errors raised when run incorrectly are as expected. So far I have been writing two separate tests: one for correct use of the code unit, one for incorrect use of the code unit. But I think this could be a cool way to tackle the problem once you are finished. I look forward to seeing the result!

Reply:
I picked it up from here: https://docs.pytest.org/en/6.2.x/example/parametrize.html#parametrizing-conditional-raising